From e597832a0cc99cb534b6be6c129f7fed90bd4a9f Mon Sep 17 00:00:00 2001
From: Shen Zhenyi <szy@rock-chips.com>
Date: Fri, 22 Jul 2016 16:04:02 +0800
Subject: [PATCH] video: rockchip: hdmi: contrast uboot and kernel resolution

When box is starting, if kernel resolution is different from uboot,
need to clear hdmi->uboot

Change-Id: Iec56862fe20dcaccc12fefae21de55b56ab2fe54
Signed-off-by: Shen Zhenyi <szy@rock-chips.com>
(cherry picked from commit 899bf65ac08492fc5aec36a1b23509baa008d409)
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
index f3e75177a9fc..2e148e23758d 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
@@ -144,6 +144,12 @@ static void hdmi_wq_set_video(struct hdmi *hdmi)
 	} else {
 		video->vic = hdmi->vic & HDMI_VIC_MASK;
 	}
+
+	if (hdmi->uboot) {
+		if ((uboot_vic & HDMI_UBOOT_VIC_MASK) != hdmi->vic)
+			hdmi->uboot = 0;
+	}
+
 	hdmi_set_lcdc(hdmi);
 	if (hdmi->ops->setvideo)
 		hdmi->ops->setvideo(hdmi, video);
-- 
2.35.3

